src/utils/point_locator_nanoflann.C

```diff
@@ -212,7 +212,17 @@ PointLocatorNanoflann::operator() (const Point & p,
 
   // We are not going to do any searching on this processor if the
   // Point doesn't fall in our (inflated) processor bounding box!
-  bool point_in_local_bbox = search_local_bbox(p);
+  // bool point_in_local_bbox = search_local_bbox(p);
+
+  // For backwards compatibility with PointLocatorTree, which did not
+  // do any processor BoundingBox check before beginning its search,
+  // the search_local_bbox() "optimization" is currently disabled. As
+  // long as _num_results isn't large, this should not slow down the
+  // Nanoflann PointLocator too much... while also making it much more
+  // likely to work the same way that the old PointLocator worked.
+  // Perhaps as a future optimization, this local bbox search approach
+  // could be enabled using a flag.
+  bool point_in_local_bbox = true;
```

Comment:
I don't see how it'll ever be correct (for the default semilocal search) to check a local bbox.  It might still be a good idea to check a semilocal bbox first on a distributed mesh?  But I don't know how Nanoflann works internally; I'd hope that any such check on our end would be redundant.

Reply:
Hmm... are you reviewing an old commit perhaps? This line has been completely removed as of the most recent version.

Reply:
Yeah; for big PRs like this I try to go commit-by-commit.  I figured this was a bug you'd already corrected, but I wanted to point out that it might be worthwhile to try this idea "halfway" rather than drop it entirely.